modules/models.py

```diff
@@ -97,6 +97,7 @@ def load_model(model_name, loader=None):
         llama_attn_hijack.hijack_llama_attention()
 
     shared.settings.update({k: v for k, v in metadata.items() if k in shared.settings})
+    shared.model_name = model_name
```

Comment:
This change is probably okay, but occurences of `shared.model_name = model_name` in other parts of the code should be removed then.